extension-androidauto/src/main/java/com/mapbox/maps/extension/androidauto/MapboxCarMap.kt

```diff
@@ -44,18 +50,24 @@ class MapboxCarMap(
    * The carContext can also be found in the [MapboxCarMapObserver] callbacks. Make sure to
    * call [MapboxCarMap.clearObservers] when your car session is destroyed.
    */
-  val carContext: CarContext = mapInitOptions.context as? CarContext
-    ?: throw IllegalArgumentException("You must construct a MapboxCarMap with the CarContext")
-
-  private val carMapSurfaceOwner = CarMapSurfaceOwner()
-  private val carMapSurfaceCallback = CarMapSurfaceCallback(
-    carContext,
-    carMapSurfaceOwner,
-    mapInitOptions
-  )
+  val carContext: CarContext by lazy {
+    carMapSurfaceOwner.carContext
+  }
 
-  init {
-    carMapSurfaceCallback.onBind()
+  /**
+   * Calls [AppManager.setSurfaceCallback] to give Mapbox access to render the map onto Android
+   * Auto head units. You should only call this once per [CarContext] in a [Session].
+   *
+   * @param carContext Android Auto CarContext
+   * @param mapInitOptions parameter to initialize the head unit map
+   */
+  fun setup(
+    carContext: CarContext,
+    mapInitOptions: MapInitOptions,
+  ) = apply {
+    mapInitOptions.verifyCarContext()
+    carMapSurfaceOwner.setup(carContext, mapInitOptions)
```

Comment:
Hmm. What if lazy [this.mapInitOptions](https://github.com/mapbox/mapbox-maps-android/pull/1603/files#diff-0175e2b342c66fe0e164006d7533334fe546e307034dc194b7351b4337376c5cR40) would be accessed before `setup` call? Would it init lazy with the outdated Options and not the mapInitOptions passed here in the `setup`?

Reply:
It will crash if you try to access mapInitOptions before a setup call. Categorizing this as a developer error, you should set up the instance before using it. I took a pass to improve these error messages.

1. When you try to construct MapboxCarMap with a regular context - must come from Android Auto
```
java.lang.IllegalStateException: You must setup the MapboxCarMap MapInitOptions with a CarContext
```

2. When you try to access car context before it has been assigned
```
java.lang.IllegalStateException: You must call MapboxCarMap.setup before you can access the MapboxCarMap.carContext
```

3. When you try to access car context before it has been assigned
```
java.lang.IllegalStateException: You must call MapboxCarMap.setup before you can access the MapboxCarMap.mapInitOptions
```

Added some unit tests to cover it
``` kotlin
  @Test(expected = IllegalStateException::class)
  fun `MapboxCarMap crashes when context is not a CarContext`() {
    val mapInitOptions = mockk<MapInitOptions> {
      every { context } returns mockk()
    }

    MapboxCarMap().setup(mockk(), mapInitOptions)
  }

  @Test(expected = IllegalStateException::class)
  fun `MapboxCarMap crashes when carContext is accessed before setup`() {
    MapboxCarMap().carContext
  }

  @Test(expected = IllegalStateException::class)
  fun `MapboxCarMap crashes when mapInitOptions is accessed before setup`() {
    MapboxCarMap().mapInitOptions
  }
```